### PR TITLE
Apply task chains to more code in FWCore/Framework

### LINKED ISSUE #209: Apply edm::waiting_task::chain to more of the FWCore/Framework code
Hopefully this change will aid in future maintenance.

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Looked for places where edm::make_waiting_task could be replaced with use of edm::waiting_task::chain constructs.

#### PR validation:

Code compiles and framework unit tests all pass.

resolves makortel/framework#209

### CODE REVIEW

File: FWCore/Framework/src/EventProcessor.cc [hunk 1]
```diff
@@ -1294,159 +1294,149 @@ namespace edm {
...
-        // Pass in iSync to let the EventSetup system know which run and lumi
-        // need to be processed and prepare IOVs for it.
-        // Pass in the endIOVWaitingTasks so the lumi can notify them when the
-        // lumi is done and no longer needs its EventSetup IOVs.
-        espController_->eventSetupForInstanceAsync(
-            iSync, queueLumiWorkTaskHolder, status->endIOVWaitingTasks(), status->eventSetupImpls());
-        sentry.completedSuccessfully();
+            status->setResumer(std::move(iResumer));
+
+            auto group = task.group();
```
Comment (Dr15Jones) [inline FWCore/Framework/src/EventProcessor.cc:1348]: NOTE: I think in a latter PR this routine should be subdivided into smaller functions which would be called by this routine.

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/src/Schedule.cc [hunk 2]
```diff
@@ -648,6 +649,18 @@ namespace edm {
...
+    template <typename F>
+    auto doCleanup(F&& iF) {
+      auto wrapped = [f = std::move(iF)](std::exception_ptr const* iPtr, edm::WaitingTaskHolder iTask) {
+        CMS_SA_ALLOW try { f(); } catch (...) {
+        }
+        if (iPtr) {
+          iTask.doneWaiting(*iPtr);
+        }
+      };
+      return wrapped;
```
Comment (makortel) [inline FWCore/Framework/src/Schedule.cc:662]: Would
```cpp
return [f = std::forward<F>(iF>)](...){...};
```
work?

**Comment by jfernan2** [comment]
please test
Retriggering to get a clean test environment, many PRs simultaneously included on top of this one

**Comment by jfernan2** [comment]
+1

**Comment by makortel** [comment]
+1

**Comment by makortel** [comment]
@cmsbuild, please test

Let's retest to be really sure

**Comment by makortel** [comment]
Just to confirm explicitly that the tests look ok.

**Comment by qliphy** [comment]
+1

**Pull request merged**